src/Dialect/ONNX/Rewrite.td

```diff
@@ -700,4 +702,14 @@ def ConstantOpNormalizationPattern6: Pat<
    [(AttributeIsNotNull:$stringsAttr)]
 >;
 
+//===----------------------------------------------------------------------===//
+// Canonicalization for ONNXLessOp
+//===----------------------------------------------------------------------===//
+
+def LessOpSameCastPattern: Pat<
+   (ONNXLessOp (ONNXCastOp $a, $to1), (ONNXCastOp $b, $to2)),
+   (ONNXLessOp $a, $b),
+   [(Equal $to1, $to2)]
+>;
```

Comment:
I see this pattern in a practical model. So add it here.

Reply:
Example: 
```
#include <iostream>
using namespace std;

int main() {
  short s = -1;
  unsigned short us = 1;

  // original
  bool res = (unsigned int)s < (unsigned int)us;
  cout << "res is " << ((res == true) ? "true" : "false") << "\n";

  // eliminate the casts
  bool res1 = s < us;
  cout << "res1 is " << ((res1 == true) ? "true" : "false") << "\n";

  return 0;
}
```
% clang++  cast.cpp; ./a.out
res is false
res1 is true
```

Reply:
@etiotto we would like to have this one merged today. Do you mind either tabling your issue, or enhancing Tung's PR, as he is probably gone for the weekend. Thanks

Reply:
We cannot merge it anyway because the Windows CI is broken...

Reply:
I see @tungld added an additional check `(HaveSameElementType $a, $b)` to the rewrite rule. In its current form the rules will kick in when:
  1. the cast target types are the same, and
  2. the data type of `$a` and `$b` are the same
  
This is not safe either, this program yields different results when the casts are removed:
```
#include <iostream>
using namespace std;

int main() {
  float f1 = -1.0f, f2 = 1.0f;

  // original
  bool res = (unsigned int)f1 < (unsigned int)f2;
  cout << "res is " << ((res == true) ? "true" : "false") << "\n";

  // eliminate the casts
  bool res1 = f1 < f2;
  cout << "res1 is " << ((res1 == true) ? "true" : "false") << "\n";

  return 0;
}
```
% clang++ cast.cpp; ./a.out
res is false
res1 is true

Reply:
> That pattern is OK because the cast target types are not unsigned && sizeof(float) == sizeof(int).

@etiotto  Added these constraints. Thanks for your comments!